temporal-sdk/src/main/java/io/temporal/client/WorkflowStartOperationUpdate.java

```diff
@@ -0,0 +1,255 @@
+  }
+
+  private static <R> Builder<R> createBuilder(Object updateFunc, Object[] args) {
+    SerializedLambda meta = JavaLambdaUtils.toSerializedLambda(updateFunc);
+    if (meta == null || meta.getImplMethodKind() != MethodHandleInfo.REF_invokeInterface) {
+      throw new IllegalArgumentException("Invalid method reference: " + updateFunc);
+    }
+
+    String updateName = null;
+    String workflowType = null;
+    Class<R> resultClass = null;
+    String updateMethodName = meta.getImplMethodName();
+    try {
+      Class<?> workflowClass = Class.forName(meta.getImplClass().replace('/', '.'));
+      workflowType = workflowClass.getSimpleName();
+      Method[] allMethods = workflowClass.getDeclaredMethods();
+      for (Method method : allMethods) {
+        if (method.getName().equals(updateMethodName)) {
+          resultClass = (Class<R>) method.getReturnType();
+          UpdateMethod annotation = method.getAnnotation(UpdateMethod.class);
+          if (annotation == null) {
+            throw new IllegalArgumentException(
+                "Method '" + updateMethodName + "' does not have UpdateMethod annotation");
+          }
+          updateName = method.getName();
+          if (!annotation.name().isEmpty()) {
+            updateName = annotation.name();
+          }
+        }
+      }
+      if (resultClass == null) {
+        throw new IllegalArgumentException(
+            "Update method with name '" + updateMethodName + "' not found on: " + workflowClass);
+      }
+    } catch (ClassNotFoundException e) {
+      throw new IllegalArgumentException("Invalid method reference", e);
+    }
+
+    return new Builder<>(
+        UpdateOptions.<R>newBuilder().setUpdateName(updateName).setResultClass(resultClass),
+        workflowType,
+        args);
+  }
+
+  private final UpdateOptions<R> options;
+
+  private final Object[] args;
+
+  private final CompletableFuture<UpdateHandle<R>> handle;
+
+  private final String workflowType;
+
+  private WorkflowStartOperationUpdate(
+      UpdateOptions<R> options, String workflowType, Object[] args) {
+    this.options = options;
+    this.args = args;
+    this.workflowType = workflowType;
+    this.handle = new CompletableFuture<>();
+  }
+
+  public UpdateOptions<R> getOptions() {
+    return options;
+  }
+
+  public Object[] getArgs() {
+    return this.args;
+  }
+
+  public String getWorkflowType() {
+    return workflowType;
+  }
+
+  @Override
+  public boolean cancel(boolean mayInterruptIfRunning) {
+    return handle.cancel(mayInterruptIfRunning);
+  }
+
+  @Override
+  public boolean isCancelled() {
+    return handle.isCancelled();
+  }
+
+  @Override
+  public boolean isDone() {
+    return handle.isDone();
+  }
+
+  public UpdateHandle<R> get() throws ExecutionException, InterruptedException {
+    return handle.get();
+  }
+
+  // TODO: hide from user?
+  public void complete(UpdateHandle<R> updateHandle) {
+    this.handle.complete(updateHandle);
+  }
```

Comment:
I don't like that this is accessible to the user, but I haven't found a way to hide it from them yet.